advanced/wallets/react-wallet-v2/src/lib/TonLib.ts

```diff
@@ -73,23 +204,29 @@ export default class TonLib {
   ): Promise<TonLib.SendMessage['result']> {
     const client = this.getTonClient(chainId)
     const walletContract = client.open(this.wallet)
-    const seqno = await walletContract.getSeqno()
+    const seqno = await retry(() => walletContract.getSeqno())
+
+    this.validateSendMessage(params)
+
     const messages = (params.messages || []).map(m => {
       const amountBigInt = typeof m.amount === 'string' ? BigInt(m.amount) : BigInt(m.amount)
       return internal({
         to: Address.parse(m.address),
+        bounce: Address.parseFriendly(m.address).isBounceable,
         value: amountBigInt,
-        body: m.payload ?? 'Test transfer from ton WalletConnect'
+        body: m.payload ? Cell.fromBase64(m.payload) : 'Test transfer from ton WalletConnect',
+        init: m.stateInit ? loadStateInit(Cell.fromBase64(m.stateInit).beginParse()) : undefined
```

Comment:
Do you think we could do something like 
```
const messages = parseTonMessages(params)
messages.map(.....)
```

Basically make the validation function return a validated, typed, version of the message so we avoid all of the duplicate parsing. 
Wyt?